README.md

```diff
@@ -71,6 +71,181 @@ You must create you ticket branches off the NCIOCPL repo such that secrets are u
 	* `package.json` and `package-lock.json` - you should know what these are.
 	* `README.md` - this document
 
+## Making API Calls
+API endpoint domain has to be set as an initialization parameter for the application in `/src/index.js`.
+
+One API Domain Multiple Endpoints:
+--
+If a call needs to be made to an endpoint for the current weather data like the Open Weather Map API (https://api.openweathermap.org/data/2.5/weather?q=London).
+The "apiEndpoint" in `/src/index.js` can be set to "https://api.openweathermap.org/data/2.5".
+Once the "apiEndpoint" is set, the service name endpoint to get current weather data can be defined in `/src/services/api/endpoints.js` using the "getEndpoints" by adding one for the current weather data to "urls".
+```
+const urls = {
+	currentWeatherData: `${API_ENDPOINT}/weather`,
+};
+```
+Since the actual call for the current weather data has a parameter "q" for the city whose weather data needs to be retrieve. That would be passed as a parameter to the action call by creating a file for the service call.
+Create a file "getCurrentWeatherDataResults.js" in `/src/services/api/actions`
+```
+import { getEndpoint } from '../endpoints';
+
+export const getCurrentWeatherDataResults = ({ q }) => {
+	const endpoint = getEndpoint('currentWeatherData');
+	return {
+		method: 'GET',
+		endpoint: `${endpoint}?q=${q}`,
+	};
+};
+```
+
+To make a call to get the current weather data from a "Dashboard" view within the application, create a "Dashboard.jsx" file in `/src/views/Dashboard`
+```
+import React, { useEffect, useState } from 'react';
+
+import { useCustomQuery } from 'src/hooks';
+import { getCurrentWeatherDataResults } from 'src/services/api/actions';
+
+const Dashboard = () => {
+	const q = "London";
+	const [doneLoading, setDoneLoading] = useState(false);
+	const [stateCurrentWeatherDataResults, setStateCurrentWeatherDataResults] = useState();
+	// Fetch results for current weather data for "London"
+	const currentWeatherDataResults = useCustomQuery(getCurrentWeatherDataResults({ q }));
+
+	useEffect(() => {
+		// Set results returned from fetch in state once fetch is done
+		// and fetch payload is defined
+		if (!currentWeatherDataResults.loading && currentWeatherDataResults.payload) {
+			setStateCurrentWeatherDataResults(currentWeatherDataResults.payload);
+			setDoneLoading(true);
+		}
+	}, [currentWeatherDataResults.loading, currentWeatherDataResults.payload]);
+
+	return (
+		<>
+            {doneLoading && ({Content to be shown once fetch is complete})}
+            {!doneLoading && ({Show some visual loading state here while fetch is still in progress...})}
+        </>
+	);
+};
+
+export default Dashboard;
+```
```

Comment:
If in the first example we show multiple endpoints being possible, I feel as though this blurb isn't necessary.

Reply:
With you on that. Will make the update.

Reply:
Updated